src/hooks/sponsorships.ts

```diff
@@ -428,7 +403,15 @@ export function useJoinSponsorshipAsOperator() {
                                 sponsorship.id,
                                 operator.id,
                             ),
-                            () => joinSponsorshipModal.pop({ sponsorship, operator }),
+                            async () => {
+                                const wallet = await (await getSigner()).getAddress()
+
+                                await joinSponsorshipModal.pop({ sponsorship, operator })
```

Comment:
The modal does not disappear until the Graph indexed block is at the newly mined block's height (mined by the joining).